src/components/twilio-voice-dialer/public/twilio-voice-dialer.js

```diff
@@ -44,29 +55,68 @@ class TwilioVoiceDialer extends HTMLElement {
     this.#call.disconnect();
   }
 
-  #handleInit() {
+  async #handleInit() {
     this.#device = new Twilio.Device(this.#token, { logLevel: 1 });
     this.#setStatus('idle');
+
+    if (Boolean(this.getAttribute('register'))) {
+      await this.#handleRegister();
+    }
+  }
+
+  async #handleRegister() {
+    await this.#device.register();
+    this.#device.on('incoming', (call) => {
+      call.on('disconnect', this.#reset.bind(this));
+      call.on('cancel', this.#reset.bind(this));
+      call.on('reject', this.#reset.bind(this));
+      this.#call = call;
+      this.#setStatus('incoming');
+    });
+
+    if (!Boolean(this.getAttribute('register'))) {
+      this.setAttribute('register', 'true');
+      this.#setStatus('idle');
+    }
+  }
+
+  #handleReject() {
+    this.#call.reject();
   }
 
   #render() {
     this.shadowRoot.innerHTML = `
       <div class="container">
         <p id="status">Status: pending</p>
+        <p id="register-status">Register: ${Boolean(
+          this.getAttribute('register')
+        )}</p>
         <input 
-            type="text"
-            placeholder="recipient"
-            id="recipient"
-            value=${this.getAttribute('recipient')}>
-				<button id="call">Call</button>
-				<button id="hangup">Hangup</button>
+					type="text"
+					placeholder="recipient"
+					id="recipient"
+					value=${this.getAttribute('recipient')}>
```

Comment:
This one should be self closing?

Reply:
Let's settle on a style guide for our HTML, I like how Google handles this kind of thing here: https://google.github.io/styleguide/htmlcssguide.html#HTML_Line-Wrapping

The indentation in this whole block is a little strange, let's fix that.